streams/src/test/java/org/apache/kafka/streams/processor/internals/StandbyTaskTest.java

```diff
@@ -177,11 +179,6 @@ public void shouldThrowLockExceptionIfFailedToLockStateDirectory() throws IOExce
 
     @Test
     public void shouldTransitToRunningAfterInitialization() {
-        EasyMock.expect(stateManager.changelogOffsets()).andStubReturn(Collections.emptyMap());
-        stateManager.registerStateStores(EasyMock.anyObject(), EasyMock.anyObject());
```

Comment:
I think there are still some misunderstandings how EasyMock works. When you call a method on the mock, this call is registered in the mock. Registration means that the method is expected to be called during the call of the method under test. When you call `EasyMock.replay(mock)` you end the registration phase. From now on the mock stores the methods actually called on it and returns the return values specified on the stubbings (i.e., `expect(mock).andReturn(...)`) when the stubbed method is called. After replaying the mock, you call the method under test. Finally, with `EasyMock.verify(mock)` you verify that only the registered method calls were performed during the call to the method under test.
In this specific case, the call `registerStateStores(EasyMock.anyObject(), EasyMock.anyObject())` was registered on the mock `stateManager` as an expected call. With `EasyMock.verify(stateManager);` it is verified that method `registerStateStores()` was called on the mock.
In your migration this verification disappeared.

Reply:
Yes this part of `EasyMock` I understand, the problem is that a lot of the tests had stubbings that were unnecessary which was being picked up being `MockitoJUnitRunner.StrictStubs.class`. Ontop of this `Easymock.verify(stateManager)` has no equivalent in mockito since its considered bad practice to just verify any interaction on a class (Mockito forces you to specify which method is being called, not just `Easymock.verify(someClass)`. Since the `Easymock.verify(someClass)` was typically placed at the bottom of the test methods so it was hard to tell what exactly was expected to mock and in some cases it was pointless.

Evidently I did some mistakes so I have put back `stateManager.registerStateStores(any(), any());` however note that `stateManager.registerStateStores()` is being called twice so I used `atLeastOnce`.